pyccel/ast/numpyext.py

```diff
@@ -171,6 +190,12 @@ class NumpyReal(PythonReal):
     1.0
     """
     __slots__ = ('_rank','_shape','_order')
+    def __new__(cls, arg):
+        if isinstance(arg.dtype, NativeBool):
+            return NumpyInt(arg)
```

Comment:
Shouldn't we also handle the case where `isinstance(arg.dtype, NativeInt)`?

Reply:
This is handled in `PythonReal.__new__` which contains:
```
def __new__(cls, arg):
        if isinstance(arg.dtype, NativeBool):
            return PythonInt(arg)
        elif not isinstance(arg.dtype, NativeComplex):
            return arg
        else:
            return super().__new__(cls)
```
Here it is necessary to handle booleans as in numpy function calls the arguments can be arrays. For other types there is no casting involved, we just return the argument so the contents of `PythonReal.__new__` are sufficient. As a result I didn't think it was necessary to duplicate the condition.

Reply:
I see, thanks.